app/grandchallenge/algorithms/models.py

```diff
@@ -770,6 +902,20 @@ def container(self):
     def output_interfaces(self):
         return self.algorithm_image.algorithm.outputs
 
+    @cached_property
+    def inputs_complete(self):
+        # check if all inputs are present and if they all have a value
+        input_interfaces = self.algorithm_image.algorithm.inputs.all()
+        existing_inputs = [
+            civ.interface for civ in self.inputs.all() if civ.has_value
+        ]
+        missing_inputs = [
+            interface
+            for interface in input_interfaces
+            if interface not in existing_inputs
```

Comment:
Might be clearer to have two sets of interfaces and then just check for equality. I think:

```python
return {civ.interface for civ in self.inputs.all() if civ.has_value} == {*self.algorithm_image.algorithm.inputs.all()}
```